pkg/report/writer_test.go

```diff
@@ -82,3 +88,111 @@ func TestResults_Failed(t *testing.T) {
 		})
 	}
 }
+
+func TestWrite(t *testing.T) {
+	tests := []struct {
+		name       string
+		setUpHook  bool
+		report     types.Report
+		options    flag.Options
+		wantOutput string
+		wantTitle  string // Expected title after function call
+		wantDesc   string // Expected description after function call
+	}{
+		{
+			name: "template with title and description",
+			report: types.Report{
+				SchemaVersion: report.SchemaVersion,
+				ArtifactName:  "test-artifact",
+				Results: types.Results{
+					{
+						Target: "test-target",
+						Vulnerabilities: []types.DetectedVulnerability{
+							{
+								VulnerabilityID: "CVE-2021-0001",
+								PkgName:         "test-pkg",
+								Vulnerability: dbTypes.Vulnerability{
+									Title:       "Test Vulnerability Title",
+									Description: "This is a test description of a vulnerability",
+								},
+							},
+						},
+					},
+				},
+			},
+			options: flag.Options{
+				ReportOptions: flag.ReportOptions{
+					Format:   types.FormatTemplate,
+					Template: "{{ range . }}{{ range .Vulnerabilities }}- {{ .VulnerabilityID }}: {{ .Title }}\n  {{ .Description }}\n{{ end }}{{ end }}",
+				},
+			},
```

Comment:
nit:
we can move that as variables before init `tests` struct.

Reply:
Which one? Template? Report?

Reply:
I think we can move both.

Reply:
Updated https://github.com/aquasecurity/trivy/pull/8585/commits/f976d2c1572be00419af9edf74a4cc837c9ced60